src/main.cpp

```diff
@@ -0,0 +1,40 @@
+#include "engine/engine.h"
+#include "graphics/renderer.h"
+
+extern void engine_config(sEngineConfiguration& config);
```

Comment:
Maybe the naming could be get_engine_config? And the config parameter is an out value, I am not sure yet we dont have a naming convention, so maybe we should discuss if we name out parameters differently like outConfig

The name change would make it simple for other devs to see what it does, which is important if they are forced to implement this function.